accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -94,20 +94,21 @@ library section below.
 
 ### Static errors
 
-We say that a type `T` is **nullable** if `Null <: T`.  This is equivalent to
-the syntactic criterion that `T` is any of:
+We say that a type `T` is **nullable** if `Null <: T` and not `T <: Object`.
```

Comment:
So, to be sure, the only types affected by this change are *T** where *T* <: `Object`.

Any other type which is <: `Object` is also not a supertype of `Null`.


This definition of nullable and non-nullable splits types into *three* groups, not four like before, where it was possible to be both nullable and non-nullable (and not potentially either).

1. non-nullable and potentially non-nullable (not nullable), T <: Object.
1. potentially nullable (not nullable) and potentially non-nullable (not nullable), not T<: Object, not Null <: T
1. nullable and potentially non-nullable (not non-nullable), not T <: Object, Null <: T.

So: to definitely (and potentially) nullable, definitely (and potentially) non-nullable, or potentially either.

That does seem to make sense.

Reply:
I could split this out further into:
  - definitely nullable (Null <: T and not T <: Object)
  - definitely non-nullable (T <: Object, not Null <: T)
  - assumed non-nullable (T <: Object)
     - the non-nullable types, plus T* where T is assumed non-nullable
  - potentially nullable (not T <: Object or Null <: T)
  - potentially non-nullable (not Null <: T or T <: Object)

I'm not sure this is useful though, so instead I'm just collapsing assumed non-nullable and non-nullable.  Does that make sense?

Reply:
If we don't need the extra cases, then let's not introduce them.
As long as this works, I'm fine with it.